src/ast/passes/semantic_analyser.cpp

```diff
@@ -514,7 +519,48 @@ void SemanticAnalyser::visit(Call &call)
       auto &expr = (*call.vargs)[i];
       func_arg_idx_ = i;
 
-      expr->accept(*this);
+      // Handling offsetof() 2nd argument
+      // Resolve the second offsetof() parameter as early as possible,
+      // otherwise, we need to do too much adaptation work in the semantic
+      // analyzer stage. For example:
+      //
+      //  struct foo {
+      //    int offsetof;   > offsetof(struct foo, offsetof)
+      //    int kstack;     > offsetof(struct foo, kstack)
+      //    int ctx;        > offsetof(struct foo, ctx)
+      //    int comm;       > offsetof(struct foo, comm)
+      //  }
+      if (func_ == "offsetof" && func_arg_idx_ == 1)
+      {
+        auto identifier = dynamic_cast<Identifier *>(call.vargs->at(1));
+        auto builtin = dynamic_cast<Builtin *>(call.vargs->at(1));
+        auto _call = dynamic_cast<Call *>(call.vargs->at(1));
```

Comment:
@danobi We must handle the builtins and calls, like `offsetof(struct foo, offsetof)` and `offsetof(struct foo, comm)`, because builtin and call will `dynamic_cast` to `nullptr`.

Reply:
Yes, good catch